Sources/Core/HardwareBreakpointManager.cpp

```diff
@@ -64,9 +65,8 @@ ErrorCode HardwareBreakpointManager::enableLocation(Site const &site,
   }
 
   enumerateThreads(thread, [&](Target::Thread *t) {
-    if (t->state() == Target::Thread::kStopped && !enabled(t)) {
-      error = enableLocation(site, idx, t);
-    }
+    DS2ASSERT(t->state() == Target::Thread::kStopped);
```

Comment:
Why did this become an assert? I see you also do it in disableLocations.
Are all threads supposed to be stopped when enabling a new hardware breakpoint?

Reply:
We discussed it a bit with stephane yesterday on messenger, but basically this function is now only run on breakpoint creation or thread creation, both of which assume the threads are stopped.

Reply:
Or at least, the thread in question.